src/motor-operations.cc

```diff
@@ -84,19 +85,97 @@ static char test_acceleration_ok(float acceleration) {
 }
 #endif
 
+struct history_slot{
+    int steps[BEAGLEG_NUM_MOTORS];
+    int max_axis_steps;
+};
+
+typedef RingQueue<struct history_slot, QUEUE_LEN> SlotHistory;
+typedef FixedArray<unsigned,BEAGLEG_NUM_MOTORS> SlotPartial;
+
+
+// This class is delegated to handle the update of the absolute
+// position in steps of each motor and it should be callbacked whenever this operations
+// has to be performed.
+class MotionQueueMotorOperations::StatusListener : public MotionQueue::EventReceiver {
+public:
+    StatusListener(MotionQueue *backend, MotorCoordSystem *current_position_steps)
+                        : backend_(backend), current_position_steps_(current_position_steps) {};
+    virtual void update_absolute_position();
+    void push_in_history(const LinearSegmentSteps &param, const int defining_axis_steps) {
+        struct history_slot *just_uploaded = execution_history.append();
+        for (int i = 0; i < MOTION_MOTOR_COUNT; ++i) {
+            just_uploaded->steps[i] = param.steps[i];
+        }
+        just_uploaded->max_axis_steps = defining_axis_steps;
+    }
+private:
+    MotionQueue *const backend_;
+
+    SlotHistory execution_history;
+    SlotPartial last_partial;
+    MotorCoordSystem *const current_position_steps_;
+};
+
+void MotionQueueMotorOperations::StatusListener::update_absolute_position() {
+    int i;
```

Comment:
Declare the variable as close as possible where it is used. Ideally within the for loop directly.